internal/integration/base/api.go

```diff
@@ -241,22 +241,27 @@ func (apiSuite *APISuite) WaitForBootDone(ctx context.Context) {
 	ctx, cancel := context.WithTimeout(client.WithNodes(ctx, nodes...), 3*time.Minute)
 	defer cancel()
 
-	apiSuite.Require().NoError(apiSuite.Client.EventsWatch(ctx, func(ch <-chan client.Event) {
-		defer cancel()
-
-		for event := range ch {
-			if msg, ok := event.Payload.(*machineapi.SequenceEvent); ok {
-				if msg.GetAction() == machineapi.SequenceEvent_STOP && msg.GetSequence() == runtime.SequenceBoot.String() {
-					delete(nodesNotDoneBooting, event.Node)
-
-					if len(nodesNotDoneBooting) == 0 {
-						return
+	err := retry.Constant(time.Minute*2, retry.WithUnits(time.Second)).Retry(func() error {
+		e := apiSuite.Client.EventsWatch(ctx, func(ch <-chan client.Event) {
+			defer cancel()
+
+			for event := range ch {
+				if msg, ok := event.Payload.(*machineapi.SequenceEvent); ok {
+					if msg.GetAction() == machineapi.SequenceEvent_STOP && msg.GetSequence() == runtime.SequenceBoot.String() {
+						delete(nodesNotDoneBooting, event.Node)
+
+						if len(nodesNotDoneBooting) == 0 {
+							return
+						}
 					}
 				}
 			}
-		}
-	}, client.WithTailEvents(-1)))
+		}, client.WithTailEvents(-1))
 
+		return retry.ExpectedError(e)
```

Comment:
I don't think we should retry _every_ error here, as it probably hides some legitimate errors?

and you said that error was about context deadline exceeded, so it won't help here?

Reply:
yeah, that's just my experiments. I will revert that back